base/linalg/arpack.jl

```diff
@@ -257,7 +257,7 @@ for (T, TR, naupd_name, neupd_name) in
     ((:Complex128, :Float64, :znaupd_, :zneupd_),
      (:Complex64,  :Float32, :cnaupd_, :cneupd_))
     @eval begin
-        function naupd(ido, bmat, n, evtype, nev, TOL::Ref{$TR}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
+        function naupd(ido, bmat, n, evtype, nev, TOL::Union{Array{$T},Ref{$T}}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
```

Comment:
I belive Tony means that the type parameter here should be `TR` and not just `T` (you can see `TR` is being used in the `ccall` below.